### PR TITLE
ref(server): Remove legacy id fallback

### PR DESCRIPTION
**Pull request by jan-auer**
Since project states are cached by project keys, we no longer need the lookup from project key to project id for the legacy store endpoint. Instead, Relay can query for the key's project state directly and then use that for normalization.

Project IDs are now optional on both the project state and in `RequestMeta`. Event normalization then writes the project ID into the Envelope's DSN and into the event payload, after obtaining it from the project config endpoint. 

Since proxy and static Relays do not obtain project configuration, they no longer support the legacy API endpoint.

### CODE REVIEW

**Comment by jan-auer** [comment]
This is still incomplete and will result in wrong normalization. The current `RequestMeta` deserializer assumes a project ID of `0` in the DSN, which will cause the event to be rejected eventually. This still needs to be fixed.

**Review by untitaker — APPROVED** [review]

File: relay-server/src/actors/project.rs [hunk 1]
```diff
@@ -87,18 +87,12 @@ pub struct LimitedProjectConfig {
     pub datascrubbing_settings: DataScrubbingConfig,
 }
 
-/// Returns an invalid placeholder project identifier.
-fn default_project_id() -> ProjectId {
-    ProjectId::new(0)
-}
-
```
Comment (RaduW) [inline relay-server/src/actors/project.rs:94]: Why is using an invalid project id (0) better/safer than using `Option<ProjectId>` ?
IMO we either always have a valid ProjectId and then there is no need for 0 or we don't and then it's cleaner to use and Option.

File: relay-server/src/actors/project.rs [hunk 2]
```diff
@@ -137,7 +131,7 @@ pub struct ProjectState {
 #[derive(Debug, Serialize)]
 #[serde(rename_all = "camelCase", remote = "ProjectState")]
 pub struct LimitedProjectState {
-    pub project_id: ProjectId,
```
Comment (RaduW) [inline relay-server/src/actors/project.rs:140]: I'm confused here we seem to be going the opposite way from ProjectId (with possible invalid value 0 ) to Option<ProjectId>.

**Review by RaduW — APPROVED** [review]

File: relay-server/src/actors/project.rs (hunk 1 above)
Comment (jan-auer) [reply relay-server/src/actors/project.rs:94]: You're right, but you're looking at code that is being removed. This PR does exactly what you propose.

**Pull request merged**